### DISCUSSION TITLE
[2024-03-04] GUI Team Demo Day

### DISCUSSION THREAD

**Discussion by farmaazon** [enso-org/enso#9265] [Weekly Demo Day]
We have one PR particularly interesting (still in progress):
![image](https://github.com/enso-org/enso/assets/3919101/94f062b9-7a5c-4cf9-98f2-a282ebc95218)

**Comment by MichaelMauderer** [comment]
* #9141 ("fixes" #8963 )
* #9186

 [Peek 2024-03-01 16-06.webm](https://github.com/enso-org/enso/assets/1428930/3f62aad8-8426-4f8f-b49e-57769b22bfb4)

* #9242

**Comment by farmaazon** [comment]
I was working mostly fixing regressions, adding more tests and fixing existing one (or the bugs they sometimes catch)
* #9127 - an interesting case of transient test failure which pointed to existing issues in our event handling.
* #9155 - it was a problem when handling with Windows paths.
* #9153 - a fix for our shortcut system, which didn't handle `click` or `pointerup` events.
* #9197 - also includes tests for selection.
* #9198 - A fix for large, unstable project of @jdunkerley.
* #9238 - Some further test fixes, and add HTML reports with trace to make debugging transient failures easier.

In the meantime, slowly progressed with Drop-down filtering.

**Comment by vitvakatu** [comment]
These weeks I was mostly progressing with two issues:
- https://github.com/enso-org/enso/pull/9093
- https://github.com/enso-org/enso/pull/9130

https://github.com/enso-org/enso/assets/6566674/d9036ce9-57a4-429b-9bd9-6392782136ea

Also I few minor things were finished just before the demo:
- https://github.com/enso-org/enso/pull/9253
- https://github.com/enso-org/enso/pull/9264

https://github.com/enso-org/enso/assets/6566674/43af88a7-2b08-4f7d-8f4a-f47b41e51b2e

**Comment by kazcw** [comment]
<table>
<tr><th>UI Feature</th><th>Before</th><th>After</th></tr>
<tr><td>Top bar (#8672, #8673)</td><td><img src="https://github.com/enso-org/enso/assets/1047859/64627a7e-5fe4-4ac4-bb18-93ee108262a0"></td><td><img src="https://github.com/enso-org/enso/assets/1047859/656d7b77-c7c0-4e67-8faa-324304efd55c"></td></tr>
<tr><td>Self-argument arrows (#9064)</td><td><img src="https://github.com/enso-org/enso/assets/1047859/ffaa60cd-ba88-4f7c-900d-41c918bc0461"></td><td><img src="https://github.com/enso-org/enso/assets/1047859/a745c373-96e3-4636-9eb3-0540a9a6a64a">
</td></tr>
<tr><td>Comments (#9162)</td><td colspan="2"><video src="https://github.com/enso-org/enso/assets/1047859/6b81f803-6c29-4a53-9c46-b5c301bcc934"/></td></tr>
<tr><td>Hide placeholders (#9208)</td><td colspan="2"><video src="https://github.com/enso-org/enso/assets/1047859/b54cfbe0-22cf-4ea5-a32a-9c109c3b0af9"/></td></tr>
<tr><td>Add-node menu button (#9219)</td><td colspan="2"><video src="https://github.com/enso-org/enso/assets/1047859/aa6cedba-ca7e-44c5-ab27-2f5d5f9421e8"/></td></tr>
<tr><td>Viz types in CB (#8570)</td><td colspan="2"><video src="https://github.com/enso-org/enso/assets/1047859/a0aa8ab9-27a4-452e-891a-4d280d074fc6"/></td></tr>
</table>

<table>
<tr><th>Enso Language Advancement</th><th>PR</th></tr>
<tr><td>Refine else-block definition</td><td><img width="16" src="https://github.com/enso-org/enso/assets/1047859/218ee58c-e461-4ebd-a6e4-e0b75ce33b0b" /> #8671</td></tr>
</table>

<table>
<tr><th>API design</th><th>Tracker</th></tr>
<tr><td>Proposed new widget-selection design</td><td>:triangular_ruler: #9145</td></tr>
</table>

<table>
<tr><th>:tada: Feature / :white_check_mark: bugfix / :clipboard: chore</th><th>PR</th></tr>
<tr><td>Show checkbox names (#9002)</td><td>:tada: #9146</td></tr>
<tr><td>Display matched aliases (Pt.1 of #9107)</td><td>:tada: #9148</td></tr>
<tr><td>Closing dropdowns (#7562, #7477)</td><td>:tada: #9206</td></tr>
<tr><td>Fix flickering when resizing visualization (#9136)</td><td>:white_check_mark: #9147</td></tr>
<tr><td>Y.Js immutable data type safety</td><td>:white_check_mark: #9217</td></tr>
<tr><td>Fix CB reloading</td><td>:white_check_mark: #9192</td></tr>
<tr><td>Quickfix for node edits</td><td>:white_check_mark: #9144</td></tr>
<tr><td>Code Editor property tests</td><td>:clipboard: #9131</td></tr>
</table>

**Comment by mwu-tow** [comment]
* Fixed notarization following the GraalVM bump.
* Updated the release template to advertise the new IDE.
* Started work on refactoring the Engine CI. Still in progresss.
  * Measuring current CI times.
  * Splitting the Engine CI into a few smaller jobs to start with.

**Comment by Frizi** [comment]
- Implemented https://github.com/enso-org/enso/issues/9161, fixed text escape/unescape to be fully inline with current parser implementation. Implemented automatic quote style change when necessary.

<img width="209" alt="image" src="https://github.com/enso-org/enso/assets/919491/fb8cd13b-a579-4295-85ef-4417cfdab8a0">

- Improved behavior of "Compiler error" popups. Now only one can be shown at a time, and it automatically closes once the code runs  successfuly.
- Improved performance of various edit scenarios when multiple nodes were affected, mostly relevant for group moving. All edits are batched both on Vue computed and Yjs transaction level.

https://github.com/enso-org/enso/assets/919491/154021fa-a460-4026-ad15-c7155729f87a

- Fixed interactions of node widths and visualizations. No more random weird node resizes or temporarily incorrect widths.
- Updated Vue to new version `3.4.*`, which improved performance and introduced a bunch of new useful features (e.g. `defineModel`).

**Comment by JaroslavTulach** [comment]
- #9245
    - need to be used by dashboard
- #9190
    - #9259

**Comment by farmaazon** [comment]
Next iteration plan highlights:
* Merge hiding placeholders and "add component" button under node.
* Finish Engine reconnection and drop-down filtering
* Finish GUI1 removal
* Add API for visualizations allowing adding nodes from them.
* File Browser widget - a simple one, using system File Browser
